include/swift/Basic/LangOptions.h

```diff
@@ -327,6 +327,10 @@ namespace swift {
     /// set to true.
     bool ExperimentalDependenciesIncludeIntrafileOnes = false;
 
+    /// Whether to enable experimental differentiable programming features:
+    /// `@differentiable` declaration attribute, etc.
+    bool EnableExperimentalDifferentiableProgramming = false;
```

Comment:
Note: this flag should be true by default.
Will update on `tensorflow` branch after feedback on https://github.com/apple/swift/pull/27446.